### PR TITLE
chore: add DropdownMenu component

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Related to https://github.com/coder/coder/issues/15617

### CODE REVIEW

File: site/src/index.css [hunk 1]
```diff
@@ -31,6 +31,10 @@
 		--border: 240 5.9% 90%;
 		--input: 240 5.9% 90%;
 		--ring: 240 10% 3.9%;
+		--muted: 240, 6%, 90%;
```
Comment (jaaydenh) [inline site/src/index.css:34]: We dont have tokens muted, popover, and accent in our design system. It should be possible to map these to the colors that exist there, https://www.figma.com/design/WfqIgsTFXN2BscBSSyXWF8/Coder-kit?node-id=1-2&t=ZPGincsj3IkIlTun-1

Comment (BrunoQuaresma) [reply site/src/index.css:34]: I see, I’ll do that. I’m just wondering how painful it is to map each one of these every time we add a new component 🥲.

Comment (jaaydenh) [reply site/src/index.css]: This does not happen for all components so overall its not been too bad so far

File: site/src/components/DropdownMenu/DropdownMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	return (
+		<span
+			className={cn("ml-auto text-xs tracking-widest opacity-60", className)}
+			{...props}
+		/>
+	);
+};
+DropdownMenuShortcut.displayName = "DropdownMenuShortcut";
+
+export {
```
Comment (jaaydenh) [inline site/src/components/DropdownMenu/DropdownMenu.tsx:182]: There is a preference to not place all exports at the bottom of the file.

File: site/src/components/DropdownMenu/DropdownMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
+import * as DropdownMenuPrimitive from "@radix-ui/react-dropdown-menu";
+import { Check, ChevronRight, Circle } from "lucide-react";
+import * as React from "react";
```
Comment (jaaydenh) [inline site/src/components/DropdownMenu/DropdownMenu.tsx:3]: I have been following the pattern of not using the * import for React for the other shadcn/ui components as we don't generally have this pattern in Coder

File: site/src/components/DropdownMenu/DropdownMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	DropdownMenuPrimitive.SubTrigger.displayName;
+
+const DropdownMenuSubContent = React.forwardRef<
+	React.ElementRef<typeof DropdownMenuPrimitive.SubContent>,
+	React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubContent>
+>(({ className, ...props }, ref) => (
+	<DropdownMenuPrimitive.SubContent
+		ref={ref}
+		className={cn(
+			"z-50 min-w-[8rem] overflow-hidden rounded-md border border-solid bg-surface-primary p-1 text-content-secondary shadow-lg data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
```
Comment (jaaydenh) [inline site/src/components/DropdownMenu/DropdownMenu.tsx:47]: For the other shadcn/ui components I have been converting very long classNames to multi line strings, roughly grouped by the general types of classes. This make it easier to read.

Comment (jaaydenh) [reply site/src/components/DropdownMenu/DropdownMenu.tsx:47]: Also, we are not currently using the classname sorting

**Review by jaaydenh — APPROVED** [review]
Just let a few comments about the style to match how to the other shadcn components have been integrated. Otherwise looks good.

File: site/src/components/DropdownMenu/DropdownMenu.tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply site/src/components/DropdownMenu/DropdownMenu.tsx:47]: We should definitely use `cn` if we want to create extendable components. It really helps in figuring out class hierarchies and merging similar styles.

> The function cn is a helper utility that makes it easier to handle className manipulation in a React + Tailwind environment, ensuring that classNames are appropriately merged and any conflicts are resolved, making the component styling more consistent and maintainable.

File: site/src/components/DropdownMenu/DropdownMenu.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+/**
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/components/DropdownMenu/DropdownMenu.stories.tsx:1]: Totally! Idk why I put it in the stories 🤦

File: site/src/components/DropdownMenu/DropdownMenu.tsx (hunk 4 above)
Comment (jaaydenh) [reply site/src/components/DropdownMenu/DropdownMenu.tsx]: Ah, I meant the biome classname sorting that just sorts the order the utility classes appear in the code.

**Pull request merged**